providers/azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/features/LocalNetworkGatewayApiLiveTest.java

```diff
@@ -0,0 +1,122 @@
+
+   @Test
+   public void createLocalNetworkGateway() {
+      AddressSpace localAddresses = AddressSpace.create(ImmutableList.of("192.168.0.0/24"));
+      LocalNetworkGatewayProperties props = LocalNetworkGatewayProperties.builder().gatewayIpAddress("1.2.3.4")
+            .localNetworkAddressSpace(localAddresses).build();
+
+      LocalNetworkGateway gw = api().createOrUpdate(name, LOCATION, null, props);
+
+      assertNotNull(gw);
+      assertEquals(gw.name(), name);
+      assertNotNull(gw.properties());
+      assertNotNull(gw.properties().gatewayIpAddress());
+      assertEquals(gw.properties().gatewayIpAddress(), "1.2.3.4");
+      assertNotNull(gw.properties().localNetworkAddressSpace());
+      assertTrue(gw.properties().localNetworkAddressSpace().addressPrefixes().contains("192.168.0.0/24"));
+   }
+
+   @Test(dependsOnMethods = "createLocalNetworkGateway")
+   public void getLocalNetworkGateway() {
+      assertNotNull(api().get(name));
+   }
+
+   @Test(dependsOnMethods = "createLocalNetworkGateway")
+   public void listLocalNetworkGateways() {
+      assertTrue(any(api().list(), new Predicate<LocalNetworkGateway>() {
+         @Override
+         public boolean apply(LocalNetworkGateway input) {
+            return name.equals(input.name());
+         }
+      }));
+   }
+
+   @Test(dependsOnMethods = "createLocalNetworkGateway")
+   public void updateLocalNetworkGateway() {
+      // Make sure the resource is fully provisioned before modifying it
+      waitUntilAvailable(name);
+      
+      LocalNetworkGateway gw = api().get(name);
+      AddressSpace localAddresses = AddressSpace.create(ImmutableList.of("192.168.0.0/24", "192.168.1.0/24"));
+      gw = api().createOrUpdate(name, LOCATION, ImmutableMap.of("foo", "bar"),
+            gw.properties().toBuilder().localNetworkAddressSpace(localAddresses).build());
+
+      assertNotNull(gw);
+      assertTrue(gw.tags().containsKey("foo"));
+      assertEquals(gw.tags().get("foo"), "bar");
+      assertTrue(gw.properties().localNetworkAddressSpace().addressPrefixes().contains("192.168.0.0/24"));
+      assertTrue(gw.properties().localNetworkAddressSpace().addressPrefixes().contains("192.168.1.0/24"));
+   }
+
+   @Test(dependsOnMethods = { "getLocalNetworkGateway", "listLocalNetworkGateways", "updateLocalNetworkGateway" }, alwaysRun = true)
+   public void deleteLocalNetworkGateway() {
+      // Make sure the resource is fully provisioned before deleting it
+      waitUntilAvailable(name);
+      URI uri = api().delete(name);
+      assertResourceDeleted(uri);
+   }
+
+   private LocalNetworkGatewayApi api() {
+      return api.getLocalNetworkGatewayApi(resourceGroupName);
+   }
+
+   private boolean waitUntilAvailable(final String name) {
+      return resourceAvailable.apply(new Supplier<Provisionable>() {
+         @Override
+         public Provisionable get() {
+            LocalNetworkGateway gw = api().get(name);
+            return gw == null ? null : gw.properties();
```

Comment:
why not using `provisioningState` ?

Reply:
That's actually what it does. The predicate accepts a `Provisionable` object which is just an interface that exposes the `provisioningState` and then uses that state to check the desired status.